crates/roc_std/src/roc_list.rs

```diff
@@ -103,6 +103,33 @@ impl<T> RocList<T> {
         self.len() == 0
     }
 
+    pub fn is_unique(&self) -> bool {
+        if let Some(storage) = self.storage() {
+            storage.is_unique()
+        } else {
+            // If there is no storage, this list is empty.
+            // An empty list is always unique.
+            true
+        }
+    }
+
+    pub fn is_readonly(&self) -> bool {
+        if let Some(storage) = self.storage() {
+            storage.is_readonly()
+        } else {
+            false
+        }
+    }
+
+    // Marks a list as readonly. This means that it will be leaked.
+    // For constants passed in from platform to application, this may be reasonable.
```

Comment:
isn't this the other way around? It's a bit of memory passed from the application to the platform?

Reply:
Oh, so this has a very specific use case in mind. I have a Rust host. It spawns a task/future/thread per request. There is a constant Str or List that I want to pass into my Roc function on every request. I do not want to allocate and deallocate on every single one of these calls. That would just be a waste.

Instead, I create a RocStr. I set it as readonly. I then share it between as many threads as I want, cloning the pointer, capacity, and length, rather than cloning the entire pointed to Str and calling malloc.

I specifically ran into this with a web server. Every request, I want to pass in the base url of my website. As a result, there was an extra malloc and free for every request that I would rather avoid if possible.

-----

If something is passed from application to host, I would generally not expect that this function would get used, but I guess you might have uses.

Reply:
oh, interesting! That makes a lot of sense.